### ISSUE TITLE
Large ConfigMaps produce "token too long" errors

### ISSUE THREAD

**Issue by philwinder** [kubernetes/kubernetes#43382]
**Kubernetes version** (use `kubectl version`):
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.0", GitCommit:"283137936a498aed572ee22af6774b6fb6e9fd94", GitTreeState:"clean", BuildDate:"2016-07-01T19:26:38Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.3", GitCommit:"029c3a408176b55c30846f0faedf56aae5992e9b", GitTreeState:"clean", BuildDate:"1970-01-01T00:00:00Z", GoVersion:"go1.7.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Cloud provider or hardware configuration**:
Minikube, 4GB RAM.
- **OS** (e.g. from /etc/os-release):
OSX
- **Kernel** (e.g. `uname -a`):
```
$ uname -a
Darwin Phils-MacBook-Pro-576.local 15.6.0 Darwin Kernel Version 15.6.0: Mon Jan  9 23:07:29 PST 2017; root:xnu-3248.60.11.2.1~1/RELEASE_X86_64 x86_64
```
**What happened**:
Use case: Use a configmap to create a grafana dashboard configuration.

When the total number of lines in the configmap exceeded somewhere around 2000 lines, kubernetes would then spurt out the error: `bufio.Scanner: token too long` whenever you perform any k8s operation on it.

**What you expected to happen**:
I expected it to work.
The file is < 100kb, so there shouldn't be any file size issues.

**How to reproduce it** (as minimally and precisely as possible):
There is an example of a failing configuration here: https://gist.github.com/philwinder/b32a4285a911ed20793943d98032f3ec
Simply download that file and try and do a `create -f` or `delete -f` on that file.

**Workaround***
You can work around this issue by splitting the config map into files less than 2k lines in length.

**Comment by jayunit100** [comment]
@kubernetes/sig-scalability-feature-requests This specific example is relevant in that i guess he's serializing a large dashboard into a config-map. both  the use case and the bug might be relevant to @kubernetes/sig-scalability-feature-requests

**Comment by philwinder** [comment]
Apologies all. I upgraded my client to the most recent version at this time, `1.5.4`, and it works. Must have been an old issue.
Thanks.

**Comment by smarterclayton** [comment]
Fix was in v1.4.0 from #19603
